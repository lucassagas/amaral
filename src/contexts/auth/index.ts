export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
