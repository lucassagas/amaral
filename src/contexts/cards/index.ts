export { CardContext, CardContextProvider } from './cards';

