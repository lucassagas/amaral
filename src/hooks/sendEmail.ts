import { api } from '@/services/api'
import { useCallback } from 'react'
// import {
//   Si
// } from '../contexts/cards/models'

const sendEmail = useCallback(async ({ data }: any): Promise<void> => {
  try {
    const { data: newData } = await api.post<any>('api/send',{
      data
    })

    console.log(newData)
  } catch(err){
    console.log(err)
  } finally {
    console.log('uepa')
  }
}, [])
