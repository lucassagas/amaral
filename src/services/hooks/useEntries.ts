import { formatDate } from '@/utils/formatDate'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

export type EntriesProps = {
  id: string
  oldQuantity: number
  newQuantity: number
  quantity: number
  updatedAt: Date | string
  product: {
    name: string
  }
  createdBy: {
    name: string
  }
}

type ApiResponse = {
  entries: EntriesProps[]
  totalEntries: number
}

async function getEntries(page: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/entries', {
    params: page,
  })

  const formatedEntries = data.entries.map((entry) => ({
    ...entry,
    updatedAt: formatDate(entry.updatedAt),
  }))

  const formatedData = {
    totalEntries: data.totalEntries,
    entries: formatedEntries,
  }

  return formatedData
}

export function useEntries(
  page: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['entries', page], () => getEntries(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
