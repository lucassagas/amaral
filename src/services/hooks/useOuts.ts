import { formatDate } from '@/utils/formatDate'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

export type OutsProps = {
  id: string
  oldQuantity: number
  newQuantity: number
  quantity: number
  updatedAt: Date | string
  product: {
    name: string
  }
  createdBy: {
    name: string
  }
}

type ApiResponse = {
  outs: OutsProps[]
  totalOuts: number
}

async function getOuts(page: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/outs', {
    params: page,
  })

  const formatedOuts = data.outs.map((out) => ({
    ...out,
    updatedAt: formatDate(out.updatedAt),
  }))

  const formatedData = {
    totalOuts: data.totalOuts,
    outs: formatedOuts,
  }

  return formatedData
}

export function useOuts(
  page: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['outs', page], () => getOuts(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
