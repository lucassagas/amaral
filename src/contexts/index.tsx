import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
