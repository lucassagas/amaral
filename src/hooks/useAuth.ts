import { AuthContext } from '@/contexts/auth'
import { AuthContextData } from '@/contexts/auth/model'
import { useContext } from 'react'

export function useAuth(): AuthContextData {
  const context = useContext(AuthContext)

  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider')
  }

  return context
}
