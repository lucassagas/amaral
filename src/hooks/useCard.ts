import { CardContext } from '@/contexts/cards'
import { CardContextData } from '@/contexts/cards/models'
import { useContext } from 'react'

export function useCard(): CardContextData {
  const context = useContext(CardContext)

  if (!context) {
    throw new Error('useCard must be used within an CardProvider')
  }

  return context
}
