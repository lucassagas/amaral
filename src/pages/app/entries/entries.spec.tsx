import { AppProvider } from '@/contexts/index'
import { useEntries } from '@/services/hooks/useEntries'
import { act, fireEvent, render, screen } from '@testing-library/react'
import { GetServerSidePropsContext } from 'next'
import Entries, { getServerSideProps } from '.'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

jest.mock('@/services/hooks/useEntries', () => ({
  useEntries: jest.fn(),
}))

const mockedEntries = {
  entries: [
    {
      createdBy: {
        name: 'Lucas SagÃ¡s',
      },
      id: '03090998-a61b-4039-a0bb-565a9d90744b',
      newQuantity: 250,
      oldQuantity: 150,
      product: {
        name: 'peixe',
      },
      quantity: 100,
      updatedAt: '04/04/2022',
    },
  ],
  totalEntries: 1,
}

const renderComponent = () => {
  return render(
    <AppProvider>
      <Entries />
    </AppProvider>
  )
}

describe('<Entries/>', () => {
  it('should render correctly', () => {
    ;(useEntries as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      data: {
        entries: mockedEntries.entries,
      },
    }))

    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should render error when return error', () => {
    ;(useEntries as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      error: true,
    }))
    renderComponent()

    expect(
      screen.getByText(/falha ao obter dados das entradas\./i)
    ).toBeInTheDocument()
  })

  it('should render loading when isLoading', () => {
    ;(useEntries as jest.Mock).mockImplementation(() => ({
      isLoading: true,
    }))
    renderComponent()

    expect(screen.getByText(/loading\.\.\./i)).toBeInTheDocument()
  })

  it('should refetch data when click in refetch button', async () => {
    ;(useEntries as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      refetch: jest.fn(),
    }))

    renderComponent()

    const button = screen.getByRole('button')

    await act(async () => {
      fireEvent.click(button)
    })

    expect(useEntries).toHaveBeenCalled()
  })

  it('should test server side props ', async () => {
    const response = await getServerSideProps({} as GetServerSidePropsContext)

    expect(response).toBeDefined()
  })
})
