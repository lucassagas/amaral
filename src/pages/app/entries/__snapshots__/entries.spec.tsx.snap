// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entries/> should render correctly 1`] = `
<div>
  <div
    class="css-1vfdvw2"
  >
    <div
      class="css-6iav15"
    >
      <h2
        class="chakra-heading css-sxvlx"
      >
        Entradas
      </h2>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-fkz59o"
          type="button"
        >
          <svg
            class="chakra-icon css-12tbk7p"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5.463 4.433A9.961 9.961 0 0 1 12 2c5.523 0 10 4.477 10 10 0 2.136-.67 4.116-1.81 5.74L17 12h3A8 8 0 0 0 6.46 6.228l-.997-1.795zm13.074 15.134A9.961 9.961 0 0 1 12 22C6.477 22 2 17.523 2 12c0-2.136.67-4.116 1.81-5.74L7 12H4a8 8 0 0 0 13.54 5.772l.997 1.795z"
              />
            </g>
          </svg>
        </button>
        <a
          class="chakra-button css-1hvb8bx"
          href="/entries/create"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11 11V5h2v6h6v2h-6v6h-2v-6H5v-2z"
                />
              </g>
            </svg>
          </span>
          Cadastrar
        </a>
      </div>
    </div>
    <div
      class="css-1j4vx6g"
    >
      <table
        class="chakra-table css-e4kp03"
        role="table"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <th
              class="css-1ednkar"
            >
              Produto
            </th>
            <th
              class="css-1ednkar"
            >
              Qtd. anterior
            </th>
            <th
              class="css-1ednkar"
            >
              Entrada
            </th>
            <th
              class="css-1ednkar"
            >
              Estoque
            </th>
            <th
              class="css-1ednkar"
            >
              Responsável
            </th>
            <th
              class="css-1ednkar"
            >
              Data
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                peixe
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                150
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                100
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                250
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                Lucas Sagás
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                04/04/2022
              </p>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="chakra-stack css-1fjzvpp"
      >
        <div
          class="css-0"
        >
          <strong>
            1
          </strong>
           -
           
          <strong>
            10
          </strong>
           
          de 
          <strong />
        </div>
        <div
          class="chakra-stack css-wta9zy"
        >
          <button
            class="chakra-button css-1jw1j19"
            disabled=""
            type="button"
          >
            1
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
